### PR TITLE
Refactor ShareSketchModal.js

### LINKED ISSUE #557: Refactor TextEditor/components/ShareSketchModal.js to a functional component
https://github.com/uclaacm/TeachLAFrontend/blob/master/src/components/TextEditor/components/ShareSketchModal.js

### PR DESCRIPTION
**Pull request by remasbashanfar**
Resolves #557

### CODE REVIEW

**Review by lumisphere902 — CHANGES_REQUESTED** [review]
The refactor looks to be mostly correct, but there are a number of lint errors. Most should be fixed automatically; let me know if you have any issues with fixing them!

**Comment by reginawang3495** [comment]
hai hai i dont think you tested this manually right? if you did then these issues wouldn't be here! lmk if you need help figuring out how to run this :)

File: src/components/TextEditor/components/ShareSketchModal.js [hunk 1]
```diff
@@ -13,50 +13,50 @@ import '../../../styles/Modals.scss';
  * ShareSketchModal is a full-screen modal that displays
  * the current program's view-only route, which the user can manually
  * copy or use the "Copy To Clipboard" Button
- * @param {Boolean} showModal modal render state
- * @param {Function} toggleModal function that toggles the modal's render state
- * @param {String} shareUrl the URL to display/copy to clipboard
```
Comment (nendow02) [inline src/components/TextEditor/components/ShareSketchModal.js:18]: We don’t have to change the comments

File: src/components/TextEditor/components/ShareSketchModal.js [hunk 2]
```diff
@@ -13,50 +13,50 @@ import '../../../styles/Modals.scss';
...
+ *  toggleModal function that toggles the modal's render state
+ *  shareUrl the URL to display/copy to clipboard
  */
-class ShareSketchModal extends React.Component {
-  state = {
-    copyStatus: 'Hit "Copy to Clipboard"!',
-  };
+const ShareSketchModal = function (props) {
+  const { shareUrl, showModal, toggleModal } = props;
+  const [str, setStr] = useState([{ copyStatus: 'Hit "Copy to Clipboard"!' }]);
```
Comment (nendow02) [inline src/components/TextEditor/components/ShareSketchModal.js:22]: Can we instead make this useState called 'copyStatus' and 'setCopyStatus' that stores a string.  You may have to also change the lines that call this later on.

**Review by nendow02 — CHANGES_REQUESTED** [review]
Looks good! Just have a few small suggestions.

**Comment by lumisphere902** [comment]
LGTM once @nendow02's comments have been addressed. :)

**Comment by reginawang3495** [comment]
@remasbashanfar Would you like to resolve conflicts and then we can merge in? Thanks!

**Comment by lumisphere902** [comment]
This has been around for a while, just cleaned it up @nendow02

**Review by nendow02 — APPROVED** [review]

**Review by lumisphere902 — APPROVED** [review]

**Pull request merged**